Purchases/Networking/Operations/CreateAliasOperation.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CreateAliasOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/18/21.
+
+import Foundation
+
+class CreateAliasOperation: NetworkOperation {
+
+    let aliasCallbackCache: CallbackCache<AliasCallback>
+    let httpClient: HTTPClient
+    let authHeaders: [String: String]
+
+    init(httpClient: HTTPClient,
+         authHeaders: [String: String],
+         aliasCallbackCache: CallbackCache<AliasCallback>) {
+        self.httpClient = httpClient
+        self.authHeaders = authHeaders
+        self.aliasCallbackCache = aliasCallbackCache
+    }
+
+    func createAlias(appUserID: String, newAppUserID: String, completion: PostRequestResponseHandler?) {
+        guard let appUserID = try? appUserID.escapedOrError() else {
+            completion?(ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let cacheKey = appUserID + newAppUserID
+        let aliasCallback = AliasCallback(key: cacheKey, callback: completion)
+        if aliasCallbackCache.add(callback: aliasCallback) == .addedToExistingInFlightList {
+            return
```

Comment:
probably not in the scope of this PR, but thoughts on logging something for duplicate requests? just thinking that it could alert users to a wonky setup

Reply:
I definitely like the idea of adding logging.

Reply:
https://app.shortcut.com/revenuecat/story/12000/log-something-for-duplicate-requests